cmake/Reprojucer.cmake

```diff
@@ -2836,6 +2831,35 @@ function(jucer_project_end)
 
   endif()
 
+  if(WIN32)
+    set(user_cmd "${JUCER_POST_EXPORT_SHELL_COMMAND_WINDOWS}")
+  else()
+    set(user_cmd "${JUCER_POST_EXPORT_SHELL_COMMAND_MACOS_LINUX}")
+  endif()
+  _FRUT_abs_path_based_on_jucer_project_dir(project_root_folder ".")
+  string(REPLACE "%%1%%" "${project_root_folder}" user_cmd "${user_cmd}")
+  if(NOT user_cmd STREQUAL "")
+    if(WIN32)
+      set(full_cmd "cmd.exe" "/c" "${user_cmd}")
+    else()
+      set(full_cmd "/bin/sh" "-c" "${user_cmd}")
+    endif()
+    if(JUCER_RUN_POST_EXPORT_SHELL_COMMANDS)
+      message(STATUS "[${JUCER_PROJECT_NAME}] Running '${user_cmd}'")
+      execute_process(COMMAND ${full_cmd}
+        TIMEOUT 10  # seconds
+        RESULT_VARIABLE cmd_result
+      )
+      if(NOT cmd_result EQUAL 0)
+        message(FATAL_ERROR "Running '${user_cmd}' failed with ${cmd_result}")
```

Comment:
> After the specified number of seconds (fractions allowed), all unfinished child processes will be terminated, and the RESULT_VARIABLE will be set to a string mentioning the “timeout”.

So if a user needs more than 10 seconds, it can modify this part of Reprojucer I guess.
Should we make this an option that defaults to 10 ?

Reply:
That 10 seconds timeout comes from Projucer:
https://github.com/WeAreROLI/JUCE/blob/1e71c07a492f01022f9064560c95c2bcd938847c/extras/Projucer/Source/ProjectSaving/jucer_ProjectSaver.h#L748-L752

I guess I should have linked to the JUCE commit in the commit message.

Reply:
I rebased and added the link to the JUCE commit in the commit message of 64866dc8604b0b018e3eb0f3bdfd20ddc86b0d7c.